### PR TITLE
BBG-59: Integrate Auth0 as login provider

### PR DESCRIPTION
**Pull request by kklaeger**

### CODE REVIEW

**Comment by kklaeger** [comment]
Sorry for this large pull request and thanks to Janis who did most of the implementation 🙂

File: betting-game-react/src/auth/Has/Has.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    Yes?: React.ComponentType;
+    No?: React.ComponentType;
+}
+
+const Has: React.FC<Props> = ({ permissions, Yes, No }) => {
+    const [hasAccess, setHasAccess] = React.useState(false);
+    const AccessGranted: React.ComponentType =
+        Yes !== undefined ? Yes : () => null;
+    const AccessDenied: React.ComponentType =
+        No !== undefined ? No : () => null;
```
Comment (mbagemihl) [inline betting-game-react/src/auth/Has/Has.tsx:16]: wtf? do we really need to use Yes and No here instead of true/false?

Comment (kklaeger) [reply betting-game-react/src/auth/Has/Has.tsx:16]: yep, the variables are called Yes and No, and depending on whether access is granted or not the component Yes or No is chosen. Usually, Yes is the component the user is trying to visit and No an error page.

**Review by mbagemihl — APPROVED** [review]

File: betting-game-react/src/auth/Has/Has.tsx (hunk 1 above)
Comment (mbagemihl) [reply betting-game-react/src/auth/Has/Has.tsx:16]: lol ok

**Pull request merged**